apis/Google.Cloud.Storage.V1/Google.Cloud.Storage.V1/UrlSigner.PostPolicyElements.cs

```diff
@@ -0,0 +1,450 @@
+
+            /// <summary>
+            /// Whether the element's value should be treated as a string or not.
+            /// Allows to decide whether it needs to be sorrounded by quotations or not.
+            /// </summary>
+            bool RepresentAsString { get; }
+        }
+
+        /// <summary>
+        /// Represents a <see cref="IPostPolicyElement"/> that supports the exact match condition operation.
+        /// </summary>
+        public interface ISupportsExact<TValue> : IPostPolicyElement
+        {
+            /// <summary>
+            /// A way to convert the values of <typeparamref name="TValue"/> to
+            /// the string representation with which to include them on the <see cref="PostPolicy"/>.
+            /// </summary>
+            /// <param name="value">The value for which to return the corresponding string representation.</param>
+            /// <returns>The string representation of the given value.</returns>
+            string ValueToString(TValue value);
+        }
+
+        /// <summary>
+        /// Represents a <see cref="IPostPolicyElement"/> that supports the range match condition operation.
+        /// </summary>
+        public interface ISupportsRange<TValue> : IPostPolicyElement
+        {
+            /// <summary>
+            /// A way to convert the values of <typeparamref name="TValue"/> to
+            /// the string representation with which to include them on the <see cref="PostPolicy"/>.
+            /// </summary>
+            /// <param name="value">The value for which to return the corresponding string representation.</param>
+            /// <returns>The string representation of the given value.</returns>
+            string ValueToString(TValue value);
+        }
+
+        /// <summary>
+        /// Represents a <see cref="IPostPolicyElement"/> that supports the starts with condition operation.
+        /// </summary>
+        public interface ISupportsStartsWith : IPostPolicyElement { }
+
+        /// <summary>
+        /// The ACL that you want to allow for the objects that will be uploaded.
+        /// </summary>
+        public sealed class Acl : ISupportsExact<string>, ISupportsStartsWith
+        {
+            /// <summary>
+            /// The default acl element.
+            /// </summary>
+            public static readonly Acl Element = new Acl();
+
+            internal Acl() { }
+
+            /// <inheritdoc/>
+            string IPostPolicyElement.ElementName => "acl";
+
+            /// <inheritdoc/>
+            bool IPostPolicyElement.RepresentAsString => true;
+
+            /// <inheritdoc/>
+            string ISupportsExact<string>.ValueToString(string value) => value;
+        }
+
+        /// <summary>
+        /// The name of the bucket that you want to allow uploads to.
+        /// Alternatively, you can include the bucket name in the <see cref="Key"/> element.
+        /// </summary>
+        public sealed class BucketName : ISupportsExact<string>
```

Comment:
This can be removed given the update to conformance tests.

Reply:
Nope, this is how a a bucket field is added. This is the implementation though and will be hidden from view.